core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -54,3 +58,40 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     Configuration.setConfiguration(null)
   }
 }
+
+object ZooKeeperTestHarness {
+
+  /**
+   * Verify that a previous test that doesn't use ZooKeeperTestHarness hasn't left behind an unexpected thread
+   */
+  @BeforeClass
+  def setUpClass() {
```

Comment:
Have you verified that this gets called by JUnit? It should, but it relies on static forwarders being added to the `ZooKeeperTestHarness` class, so that's why I'm asking.

Also, it's worth adding a comment stating that this relies on the fact that the Core tests (which are the only ones that use `ZooKeeperTestHarness`) never use @BeforeClass. If they did, then the order would be unspecified and we'd have to find a different solution.